### PR TITLE
ENH: Increase coverage for miscellaneous classes

### PR DESCRIPTION
**Pull request by jhlegarreta**
- ENH: Remove unnecessary/unused include statements
- STYLE: Use the `itkNameOfTestExecutableMacro` macro in tests 
- STYLE: Prefer using `ITK_TRY_EXPECT_NO_EXCEPTION` macro in tests 
- STYLE: Improve style in tests
- ENH: Increase coverage for miscellaneous classes
- BUG: Set mesh to join facet function in split facet function test
- BUG: Fix Superclass name in RTTI macro

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Following #3022 the superclass name is no longer checked, and thus `itkPointSetTest.cxx` was not failing:
https://open.cdash.org/viewTest.php?onlypassed&buildid=7804517

even prior to 101c262.

**Comment by jhlegarreta** [comment]
Following 78189c6, wondering whether there are tools out there to flag unnecessary/unused `include` statements. I think it would be great if there were and we could add them as a pre-commit hook or to our set of linters.

**Comment by seanm** [comment]
> wondering whether there are tools out there to flag unnecessary/unused `include` statements.

There sure are. For example: https://include-what-you-use.org/

File: Modules/Core/Common/include/itkPointSet.h [hunk 1]
```diff
@@ -94,7 +94,7 @@ class ITK_TEMPLATE_EXPORT PointSet : public DataObject
   itkNewMacro(Self);
 
   /** Standard part of every itk Object. */
-  itkTypeMacro(PointSet, Object);
+  itkTypeMacro(PointSet, DataObject);
```
Comment (thewtex) [inline Modules/Core/Common/include/itkPointSet.h:97]: :eagle:

**Review by thewtex — APPROVED** [review]
:green_heart:

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**